src/test/java/com/danielfrak/code/keycloak/providers/rest/LegacyProviderTest.java

```diff
@@ -218,15 +218,35 @@ void getUserByIdShouldThrowException() {
     }
 
     @Test
-    void removeFederationLinkWhenCredentialUpdates() {
+    void removeFederationLinkWhenCredentialUpdatesAndDeleteMigrationRecord() {
         var input = mock(CredentialInput.class);
+        final String username = "user";
+
+        when(legacyUserService.removeByIdentifier(username))
+                .thenReturn(true);
+        when(userModel.getUsername())
+                .thenReturn(username);
         when(userModel.getFederationLink())
                 .thenReturn("someId");
-
+        MultivaluedHashMap<String, String> config = new MultivaluedHashMap<>();
+        config.put(DELETE_MIGRATION_RECORD, List.of("true"));
+        when(model.getConfig()).thenReturn(config);
         assertFalse(legacyProvider.updateCredential(realmModel, userModel, input));
+        assertTrue(legacyUserService.removeByIdentifier(username));
+    }
 
-        verify(userModel)
-                .setFederationLink(null);
+    @Test
+    void skipDeleteMigrationRecord() {
```

Comment:
Like I mentioned above, I think `migration record` doesn't work well as a name for the legacy user. Additionally, the name is missing some information, so I propose naming it `skipDeletingMigratedLegacyUserWhenConfigOptionIsDisabled`.